core/src/test/scala/unit/kafka/server/FetchSessionTest.scala

```diff
@@ -967,6 +967,113 @@ class FetchSessionTest {
       .setErrorCode(errorCode)
   }
 
+  @Test
```

Comment:
Is this replacing `testUpdatedPartitionResolvesId`? This is definitely cleaner, but I'm not sure we are covering the same cases here. For context, the test I mentioned before is testing different update scenarios (I probably named it poorly). Mostly the idea is that the update method works correctly. (ie, we update a partition that once had a topic ID to one that does not, etc). Maybe that is covered in some of the other tests I've added (like `def maybeUpdateRequestParamsOrName`) and we can just remove that test. What do you think? I'll also think about this a bit more.

Reply:
Alternatively I can just rewrite this.

Reply:
I wrote that test to illustrate how we could improve the readability. My concern is that they are so many lines/assertions in `testUpdatedPartitionResolvesId` and `testToForgetCases` that we get distracted and we have almost missed the most important assertions - the ones which validate what the session contains (`assertPartitionsOrder`). `assertPartitionsOrder` is actually the piece which ensures that the names are resolved or not, right?

Reply:
There are two places it may be resolved -- either in the update method if the partition with the new ID is sent in the request or in the assertPartitionsOrder. I was also trying to ensure the correct error messages are returned in the response specifically via `updateAndGenerateResponseData`, but maybe we don't care about this here?

Reply:
Right. The question is how to validate that the first update method works? You have to get the partitions from the session as well, isn't it?

Reply:
I suppose so. I wonder if we should even include the update method at all then...

Reply:
What do you mean?

Reply:
If we always resolve when iterating through the partitions, then do we need to resolve via the update method?

Reply:
Yeah, that is not really necessary as you said. I don't mind if you remote it.